server/src/main/java/com/thoughtworks/go/server/service/ElasticAgentPluginService.java

```diff
@@ -237,14 +244,15 @@ private void logToJobConsole(JobIdentifier identifier, String message) {
     }
 
     public boolean shouldAssignWork(ElasticAgentMetadata metadata, String environment, ElasticProfile elasticProfile, ClusterProfile clusterProfile, JobIdentifier identifier) {
-        Map<String, String> clusterProfileProperties = clusterProfile != null ? clusterProfile.getConfigurationAsMap(true) : Collections.EMPTY_MAP;
-        GoPluginDescriptor pluginDescriptor = pluginManager.getPluginDescriptorFor(metadata.elasticPluginId());
-        Map<String, String> configuration = elasticProfile.getConfigurationAsMap(true);
-
-        if (!StringUtils.equals(clusterProfile.getPluginId(), metadata.elasticPluginId())) {
+        if (clusterProfile == null || !StringUtils.equals(clusterProfile.getPluginId(), metadata.elasticPluginId())) {
             return false;
         }
 
+        resolveSecrets(clusterProfile, elasticProfile);
```

Comment:
What happens when secret resolution fails and what should we do on failure?

Reply:
The `shouldAssignWork` is called when a elastic agent registers with GoCD. In case, secret resolution fails, the agent will not be assigned any job nd will remain idle.
We can catch them nd ignore it.